azavea/raster-vision
# Implement experimental AWS SageMaker support

## Overview

Supersedes #1871.

This PR adds the `rastervision_aws_sagemaker` module which allows running RV jobs on AWS SageMaker in much the same way that `rastervision_aws_batch` allows running jobs on AWS Batch. 

To use this functionality, you can add something like the following to your `~/.ratervision/default` configuration file:
```ini
[SAGEMAKER]
exec_role=AmazonSageMakerExecutionRole
cpu_image=123.dkr.ecr.us-east-1.amazonaws.com/raster-vision
cpu_instance_type=ml.p3.2xlarge
gpu_image=123.dkr.ecr.us-east-1.amazonaws.com/raster-vision
gpu_instance_type=ml.p3.2xlarge
use_spot_instances=yes
```
where
- `exec_role` is an IAM role with appropriate S3 and SageMaker permissions.
- `*_image` is the URI of a docker image.
- `*_instance_type` is the instance type. See [here](https://docs.aws.amazon.com/sagemaker/latest/dg/notebooks-available-instance-types.html) for a list.

### Checklist


### Notes

- This should be considered very early-stage and experimental.
- It is currently not included in the default `pip install rastervision`.
- We are not currently making use of SageMaker features like hyperparameter tuning jobs and model deployment.

## Testing Instructions

* See new unit tests.
* Run an example with `rastervision run sagemaker ..` instead of `rastervision run batch ..`.
